packages/wallet-sdk/src/CoinbaseWalletProvider.ts

```diff
@@ -43,77 +41,46 @@ export class CoinbaseWalletProvider extends EventEmitter implements ProviderInte
     this.signer = signerType ? this.initSigner(signerType) : null;
   }
 
-  public async request<T>(args: RequestArguments): Promise<T> {
+  public async request(args: RequestArguments): Promise<unknown> {
```

Comment:
deprecating the concept of `MethodCategory`.
now, provider will check raw args.method directly and handle requests in 3 cases:
- handshake
- chain id requests, to keep the original behavior of returning mainnet as fallback
- all other requests will be passed to signer if exists. otherwise, throw error.